### PR TITLE
Use docker_bundle rule from new rules_docker repo

### PR DESCRIPTION
**Pull request by ixdy**
**What this PR does / why we need it**: switched to using the new `docker_bundle` rule from `rules_docker` instead of my patched `docker_build` rule. This also brings in some fixes for the docker rules that were missing from my fork.

Additionally, I switched out the `git_repository` rules for `http_archive` rules, since that seems to be recommended by the bazel docs (and might be faster). 

Lastly, I updated the `pkg_tar` rules to use my patch, which doesn't prepend `./` to files inside the tarballs.
This one should likely be merged upstream in the near future.

I think this is the last of the changes necessary to have `bazel run //:ci-artifacts` working properly to support using bazel for e2e in CI.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by ixdy** [comment]
We should probably review/merge https://github.com/kubernetes/repo-infra/pull/17 and update that `WORKSPACE` dependency before requesting top-level approval.

File: build/BUILD [hunk 1]
```diff
@@ -63,33 +63,34 @@ DOCKERIZED_BINARIES = {
...
     symlinks = {
         # Some cluster startup scripts expect to find the binaries in /usr/local/bin,
         # but the debs install the binaries into /usr/bin.
         "/usr/local/bin/" + binary: "/usr/bin/" + binary,
     },
 ) for binary, meta in DOCKERIZED_BINARIES.items()]
 
+[docker_bundle(
+    name = binary,
+    images = {"gcr.io/google_containers/%s:{STABLE_DOCKER_TAG}" % binary: binary + "-internal"},
```
Comment (mikedanese) [inline build/BUILD]: {STABLE_DOCKER_TAG}??? Does that actually work?

**Comment by mikedanese** [comment]
delete build/foo.tar

File: build/BUILD (hunk 1 above)
Comment (ixdy) [inline build/BUILD]: yep, with https://github.com/bazelbuild/rules_docker/pull/17 :)

**Comment by ixdy** [comment]
ha, I have no idea how that even got there! :)

File: WORKSPACE [hunk 2]
```diff
@@ -1,27 +1,42 @@
...
-    commit = "684e550a2f006dbe3cf3b3d481d3f19217b228f7",
-    remote = "https://github.com/kubernetes/repo-infra.git",
+    sha256 = "1d146ccc034d3a9d8f5f02237144b78be135a0fb06935b37c745a2e7fcfecfed",
+    strip_prefix = "repo-infra-684e550a2f006dbe3cf3b3d481d3f19217b228f7",
+    urls = ["https://github.com/kubernetes/repo-infra/archive/684e550a2f006dbe3cf3b3d481d3f19217b228f7.tar.gz"],
 )
 
-git_repository(
+# This contains a patch to not prepend ./ to tarfiles produced by pkg_tar.
+# When merged upstream, we'll no longer need to use ixdy's fork.
```
Comment (mikedanese) [inline WORKSPACE]: Can you link the CL that needs to merge here?

Comment (ixdy) [inline WORKSPACE]: done

**Pull request merged**